### PR TITLE
Move source files to confirm to Kotlin Multiplatform Gradle plugin

### PR DESCRIPTION
**Pull request by bnorm**
Besides moving sources files, clean up the Gradle build scripts to
remove redundant declarations and help IntelliJ properly configure
inter-project dependencies.

Closes: #584

### CODE REVIEW

File: okio/build.gradle [hunk 1]
```diff
@@ -35,52 +26,47 @@ kotlin {
...
     jvmTest {
-      kotlin.setSrcDirs(['jvm/src/test/java'])
       dependencies {
         implementation deps.test.junit
         implementation deps.test.assertj
         implementation deps.kotlin.test.jdk
       }
     }
+    test {
+      // bnorm: not required, but helps IntelliJ configure inter-project dependencies
```
Comment (JakeWharton) [inline okio/build.gradle]: Link to bug, remove name

File: okio/build.gradle [hunk 2]
```diff
@@ -1,17 +1,8 @@
...
-    // gonna keep it inside targets {}
-    fromPreset(presets.jvmWithJava, 'jvm') {
-      project.sourceSets.main.java.setSrcDirs(['jvm/src/main/java'])
-      project.sourceSets.test.java.setSrcDirs(['jvm/src/test/java'])
-      project.sourceSets.main.resources.setSrcDirs(['jvm/src/main/resources'])
-      project.sourceSets.main.kotlin.setSrcDirs(['jvm/src/main/kotlin'])
-      project.sourceSets.test.kotlin.setSrcDirs(['jvm/src/test/kotlin'])
-    }
-  }
+  // egor: jvmWithJava is getting deprecated but it's not clear what the replacement is.
```
Comment (JakeWharton) [inline okio/build.gradle]: Link to bug https://youtrack.jetbrains.com/issue/KT-26256, remove name

File: samples/build.gradle [hunk 3]
```diff
@@ -21,3 +17,8 @@ kotlin {
     }
   }
 }
+
+// bnorm: not required, but helps IntelliJ configure inter-project dependencies
```
Comment (JakeWharton) [inline samples/build.gradle]: Link to bug, remove name

File: samples/build.gradle [hunk 4]
```diff
@@ -1,12 +1,8 @@
 apply plugin: 'org.jetbrains.kotlin.multiplatform'
 
 kotlin {
-  targets {
-    fromPreset(presets.jvmWithJava, 'jvm') {
-      project.sourceSets.main.java.setSrcDirs(['src/main/java'])
-      project.sourceSets.test.java.setSrcDirs(['src/test/java'])
-    }
-  }
+  // egor: jvmWithJava is getting deprecated but it's not clear what the replacement is.
```
Comment (JakeWharton) [inline samples/build.gradle]: Link to bug https://youtrack.jetbrains.com/issue/KT-26256, remove name

**Comment by Egorand** [comment]
A bit off-topic: @JakeWharton is Okio's Gradle setup consistent to SQLDelight? Working on migrating Wire and wondering which project works better as a template.

**Comment by JakeWharton** [comment]
This PR brings them pretty close together. I'm not sure if there are meaningful differences anymore.

**Comment by Egorand** [comment]
Sweet

**Pull request merged**